ADR/0006-log-conventions.md

```diff
@@ -0,0 +1,196 @@
+
+Use the key `namespace` when logging the namespace of the *targeted resource* that is being modified
+or interacted with. (Note that the `namespace_name` key is automatically added by the fluent
+collectors and reflects the namespace in which the controller is running. See section on
+automatically added logs below.)
+
+This should appear in the `ADDITIONAL_JSON_DATA` at the end of the log line and can optionally
+appear in the human-readable `MESSAGE`.
+
+### 5. Where it came from?
+
+**Included in:** `MESSAGE`, `ADDITIONAL_JSON_DATA`
+
+Use the key `kind` with possible resource kind values determined by the component team.  For example, for HAS this can be `CDQ`, `Application`, `Component`, etc.
+
+Use the key `resource` with the name of the resource being acted upon.
+
+Optionally, use the key `source` to direct developers to the source code where the action occurred.
+
+Include these in the `ADDITIONAL_JSON_DATA` at the end of the log line. They can optionally appear
+in the human-readable `MESSAGE`.
+
+### Controller audit logs
+
+**Included in:** `ADDITIONAL_JSON_DATA`
+
+For the specific types of audit logs required by SSML.PW.5.1.4 Perform Event Logging, one of the key-value pairs in the log entry should be `audit: true`. This makes it easier to query aggregated logs to find these special log entries.
+
+More details can be found in [SSML-8093](https://issues.redhat.com/browse/SSML-8093), and a good example of a working implementation can be found in the [GitOps code](https://github.com/redhat-appstudio/managed-gitops/blob/c962ae99ec50e273c8cdf90d8f3a07f7a8944dc5/backend-shared/util/log.go#L28) implemented for [this story](https://issues.redhat.com/browse/GITOPSRVCE-186).
+
+### Automatically added fields
+
+Note that fluentd and Fluent Bit will annotate your log messages with the following information automatically:
+
+* `namespace_name`
+* `container_name`
+* `pod_name`
+* `container_image`
+* `pod_ip`
+* `host`
+* `hostname`
+* `namespace_labels`
+* `message`
+* `level`
+* `time`
+* and more.
+
+The cluster, node, pod and container names are also part of the log stream name.  For example, under `/aws/containerinsights/<Cluster_Name>/application`:
+
+* Fluent Bit optimized configuration sends logs to `<kubernetes-nodeName>-application.var.log.containers.<kubernetes-podName>_<kubernetes-namespace>_<kubernetes-container-name>-<kubernetes-containerID>`
+* Fluentd sends logs to `<kubernetes-podName>_<kubernetes-namespace>_<kubernetes-containerName>_<kubernetes-containerID>`
+
+For more details on Fluentd vs. Fluent Bit logs, see [Set up Fluent Bit as a DaemonSet to send logs to CloudWatch Logs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Container-Insights-setup-logs-FluentBit.html).
+
+It is best to avoid using the same key as the log collectors in our controller logs, to avoid confusion.
+
+### TaskRun logs
+
+TaskRun logs are specifically out of scope of this ADR. Those logs are typically meant to be viewed
```

Comment:
Should any logs related to the running application be called out here as well? Application logs generated after it's deployed would contain info that is meant to be viewed and should be human-readable. @jgwest @jannfis does that sound right to you?

Reply:
Agreed, let's call it out of scope. In practice, we can't force the user's app's logs to conform to our scheme. I don't think it is desired, either.

Reply:
Added in 15ac904cc11502c2e4bf59eec60f334b9b08749c.